torch/csrc/distributed/c10d/GroupRegistry.cpp

```diff
@@ -10,12 +10,11 @@ namespace {
 class GroupRegistry {
  public:
   void register_group(
-      std::string group_name,
-      // NOLINTNEXTLINE(performance-unnecessary-value-param)
+      const std::string& group_name,
```

Comment:
Why does this? We can avoid a copy sometime with the moving.

Reply:
Because group_name is used after move in the assert..